File: python/ev3dev/eg/port.py
# -*- coding: utf-8 -*-

import sys
from ev3dev import *

if __name__ == '__main__':
    print 'Waiting the EV3 brick online...'
    if ev3_init() < 1: sys.exit( 1 )

    print '*** ( EV3 ) Hello! ***'
    ev3_port_init()

    print 'Found ports:'
    for i in range( PORT_DESC__LIMIT_ ):
        type_inx = ev3_port_desc_type_inx( i )
        if type_inx != PORT_TYPE__NONE_:
            print '  type =', ev3_port_type( type_inx )
            print '  port =', ev3_port_port_name( i )
            ok, mode = get_port_mode( i, 256 )
            if ok:
                print '  mode =', mode

    sn = ev3_search_port( INPUT_2, EXT_PORT__NONE_ )

    print 'Port "%s"' % ( ev3_port_name( INPUT_2, EXT_PORT__NONE_, 0 ))
    ok, s = get_port_modes( sn, 256 )
    if ok:
        print 'Modes:', s
    else:
        print 'ERROR: get_port_modes()'

    ev3_uninit()

File: eg/port/port.c
/*
     ____ __     ____   ___    ____ __         (((((()
    | |_  \ \  /   ) ) | |  ) | |_  \ \  /  \(@)- /
    |_|__  \_\/  __)_) |_|_/  |_|__  \_\/   /(@)- \
                                               ((())))
 *//**
 *  \file  port.c
 *  \brief  ev3dev-c Ports example.
 *  \author  Vitaly Kravtsov (in4lio@gmail.com)
 *  \copyright  See the LICENSE file.
 */

#include <stdio.h>
#include "ev3.h"
#include "ev3_port.h"
#include "ev3_tacho.h"

// WIN32 /////////////////////////////////////////
#ifdef __WIN32__

#include <windows.h>

// UNIX //////////////////////////////////////////
#else

#include <unistd.h>
#define Sleep( msec ) usleep(( msec ) * 1000 )

//////////////////////////////////////////////////
#endif

int main( void )
{
	char s[ 256 ];
	int i;
	uint8_t sn;

	printf( "Waiting the EV3 brick online...\n" );
	if ( ev3_init() < 1 ) return ( 1 );

	printf( "*** ( EV3 ) Hello! ***\n" );
	ev3_port_init();

	printf( "Found ports:\n" );
	for ( i = 0; i < DESC_LIMIT; i++ ) {
		if ( ev3_port[ i ].type_inx != PORT_TYPE__NONE_ ) {
			printf( "  type = %s\n", ev3_port_type( ev3_port[ i ].type_inx ));
			printf( "  port = %s\n", ev3_port_port_name( i, s ));
			if ( get_port_mode( i, s, sizeof( s ))) {
				printf( "  mode = %s\n", s );
			}
		}
	}
	sn = ev3_search_port( INPUT_2, EXT_PORT__NONE_ );
	printf( "Port '%s'\n", ev3_port_name( INPUT_2, EXT_PORT__NONE_, 0, s ));
	if ( get_port_modes( sn, s, sizeof( s ))) {
		printf( "Modes: %s\n", s );
	} else {
		printf( "ERROR: get_port_modes()\n" );
	}
	ev3_uninit();
	printf( "*** ( EV3 ) Bye! ***\n" );

	return ( 0 );
}
